sync/src/synchronizer/get_headers_process.rs

```diff
@@ -59,6 +59,10 @@ impl<'a> GetHeadersProcess<'a> {
             let state = self.synchronizer.shared.state();
             if let Some(flag) = state.peers().get_flag(self.peer) {
                 if flag.is_outbound || flag.is_whitelist || flag.is_protect {
+                    // Remove genesis hash on locators, because `try_update_best_known_with_unknown_header_list` can use
+                    // genesis hash on headermap search, it will case some peers can not set best knonwn header and causes
+                    // the ibd download source to be lost
+                    block_locator_hashes.pop();
```

Comment:
I don't think we need to remove the genesis hash from the locator since we changed the code to loading from db first in this PR

Reply:
`get_header_view` will try searching from headermap first, if can't find it, failback to search DB.